geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/KeyHashUtil.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.redis.internal.data;
+
+import static org.apache.geode.redis.internal.RegionProvider.REDIS_SLOTS;
+
+import org.apache.geode.redis.internal.executor.cluster.CRC16;
+
+/**
+ * Utility for parsing the hashtags in a redis key and computing the redis slot of the key.
+ */
+public class KeyHashUtil {
+
+  /**
+   * Compute the slot of a given redis key
+   *
+   * @return the slot, a value between 0 and 16K.
+   */
+  public static int slotForKey(byte[] value) {
+    int startHashtag = Integer.MAX_VALUE;
+    int endHashtag = 0;
+
+    for (int i = 0; i < value.length; i++) {
+      if (value[i] == '{' && startHashtag == Integer.MAX_VALUE) {
```

Comment:
In the `StringBytesGlossary` class, there's a comment (originally in the Coder class before the constants were pulled out into their own class) stating:
> Do not use '' <-- java primitive chars. Redis uses `{@link Coder#CHARSET}` encoding so we should
not risk java handling char to byte conversions, rather just hard code `{@link Coder#CHARSET}`
chars as bytes
Is this a concern here?

Reply:
I put a couple of constants in StringBytesGlossary.

However, I don't think it is necessary for that class to hard code the decimal value of these chars ( eg 56, 48, etc.). `char` literals in java are always UTF-16. So converting them to a byte will always have the same value. That is not true for String.getBytes(), which uses a OS platform specific encoding.

So what I changed is that the constants are in still StringBytesGlossary but I changed that class to use char literals, which I think makes it easier to read and maintain.

Reply:
Ah, thanks for the explanation. Yeah, that definitely makes the class more readable. If it's not too much trouble, there are some other single-char constants in that file starting on line 243 that could also be simplified in this way.

Reply:
Done!